internal/storage/bucket_handle.go

```diff
@@ -252,3 +252,65 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 	listing = &list
 	return
 }
+
+func (b *bucketHandle) UpdateObject(ctx context.Context, req *gcs.UpdateObjectRequest) (o *gcs.Object, err error) {
+
+	obj := b.bucket.Object(req.Name)
+
+	if req.Generation != 0 {
+		obj = obj.Generation(req.Generation)
+	}
+
+	if req.MetaGenerationPrecondition != nil {
+		obj = obj.If(storage.Conditions{MetagenerationMatch: *req.MetaGenerationPrecondition})
+	}
+
+	updateQuery := storage.ObjectAttrsToUpdate{}
+
+	if req.ContentType != nil {
+		updateQuery.ContentType = *req.ContentType
+	}
+
+	if req.ContentEncoding != nil {
+		updateQuery.ContentEncoding = *req.ContentEncoding
+	}
+
+	if req.ContentLanguage != nil {
+		updateQuery.ContentLanguage = *req.ContentLanguage
+	}
+
+	if req.CacheControl != nil {
+		updateQuery.CacheControl = *req.CacheControl
+	}
+
+	if req.Metadata != nil {
+		updateQuery.Metadata = make(map[string]string)
+		for key, element := range req.Metadata {
+			if element != nil {
+				updateQuery.Metadata[key] = *element
+			}
+		}
+	}
+
+	attrs, err := obj.Update(ctx, updateQuery)
+
+	if err != nil {
```

Comment:
add a comment why this conversion is required.

Reply:
fixed